### PR TITLE
do not rely on version from Manifest

### PR DESCRIPTION
**Pull request by wolframhaussig**
solves #390

- added generated class Pf4jVersion to contain the current version
- added templating-maven-plugin to add project version into Pf4jVersion
- added UnitTest for version

### CODE REVIEW

**Comment by decebals** [comment]
It looks good. I prefer to rename `Pf4jVersion` to avoid duplication in `Pf4jVersion.VERSION` and to be able to put other useful information in the future (let's say `BUILD_NUMBER` as an example).
@wolframhaussig what do you think?

File: pf4j/src/test/java/org/pf4j/AbstractPluginManagerTest.java [hunk 1]
```diff
@@ -45,5 +46,10 @@ public void getExtensionsByType() {
         List<TestExtensionPoint> extensions = pluginManager.getExtensions(TestExtensionPoint.class);
         assertEquals(1, extensions.size());
     }
+    @Test
```
Comment (decebals) [inline pf4j/src/test/java/org/pf4j/AbstractPluginManagerTest.java:49]: please, add an empty line between methods

**Review by decebals — CHANGES_REQUESTED** [review]

**Comment by wolframhaussig** [comment]
@decebals You are right. Should I rename the class `Pf4jVersion` to `Pf4jInfo`?

**Comment by decebals** [comment]
> @decebals You are right. Should I rename the class `Pf4jVersion` to `Pf4jInfo`?

I think that `Pf4jInfo` is a better name.

File: pf4j/src/main/java-templates/org/pf4j/Pf4jVersion.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pf4j;
+
+/**
+ * class provides access to the current Pf4j version which might otherwise not be possible(e.g. in Uber-Jars where the Manifest wass overridden)
```
Comment (decebals) [inline pf4j/src/main/java-templates/org/pf4j/Pf4jVersion.java:19]: I prefer to start with upper case the sentence. I try not to be annoying :smile:.

**Comment by wolframhaussig** [comment]
@decebals I renamed the class and fixed the javadoc so feel free to have a look.

**Pull request merged**